src/v/archival/ntp_archiver_service.cc

```diff
@@ -72,6 +77,67 @@ ntp_archiver::ntp_archiver(
       _start_term);
 }
 
+void ntp_archiver::run_upload_loop() {
+    // NOTE: not using ssx::spawn_with_gate_then here because we want to log
+    // inside the gate (so that _rtclog is guaranteed to be alive).
+    ssx::spawn_with_gate(_gate, [this] {
+        return ss::with_scheduling_group(
+                 _upload_sg, [this] { return upload_loop(); })
```

Comment:
now that upload loop is per ntp_archiver, does there need to be / is there protection from multiple instances of upload loop running? not that there should be, was just thinking about defensive measures.

Reply:
Good idea, I can add some asserts to ensure that `run_upload_loop` is one-shot.